src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -177,12 +187,48 @@ async def _run_impl(self) -> None:
                 else:
                     self._buffer.update(sample)
 
+                # count the number of samples and send out a trigger when it matches
+                # the number of samples to wait for.
+                received_samples_count += 1
+                if self._wait_for_num_samples != 0:
+                    if received_samples_count == self._wait_for_num_samples:
+                        received_samples_count = 0
+                        await wait_for_samples_sender.send(None)
```

Comment:
Since it's free and even when the user should know it beforehand anyway, we could also send the number of samples received instead of `None`, maybe it could become a handy shortcut.
```suggestion
                        await wait_for_samples_sender.send(received_samples_count)
                        received_samples_count = 0
```